.github/workflows/test-build-push.yml

```diff
@@ -0,0 +1,125 @@
+name: Run tests, build image and push to GitHub Container Registry
```

Comment:
Any chance we can follow the same pattern / split as in `poktroll`?

Specifically, we have a `main-build.yml`, a `release-artifacts.yml` a `run-tests.yml`.

I realize it's a little overkill, but don't think it should be too much overhead here.

https://github.com/pokt-network/poktroll/tree/main/.github/workflows

Reply:
We can follow this pattern for the sake of consistency when it comes to files and naming convention. I'll take a look at the files and see what we can adapt (I've already taken parts of them).

Reply:
Updated to reflect the suggested pattern